### PR TITLE
[DML EP] Add GroupNorm

### PR DESCRIPTION
**Pull request by PatriceVignola**
Comparison between the different normalization operations:

### CODE REVIEW

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorGroupNorm.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+        std::vector<DML_TENSOR_DESC> inputDescs = GetDmlInputDescs();
+        std::vector<DML_TENSOR_DESC> outputDescs = GetDmlOutputDescs();
+
+        const float epsilon = kernelCreationContext.GetOptionalAttribute<float>(AttrName::Epsilon, DefaultEpsilon);
+        const bool activation = gsl::narrow_cast<bool>(kernelCreationContext.GetAttribute<int64_t>(AttrName::Activation));
+        const uint32_t groups = gsl::narrow_cast<uint32_t>(kernelCreationContext.GetAttribute<int64_t>(AttrName::Groups));
+
+        ML_CHECK_VALID_ARGUMENT(m_inputTensorDescs.size() == 3);
+
```
Comment (sumitsays) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorGroupNorm.cpp:27]: [nit] DmlOperator::Initialize will create (m_inputTensorDescs and m_outputTensorDescs) of same size as kernelCreationContext.GetInputCount(). Any specific reason to verify this again?

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp [hunk 2]
```diff
@@ -463,6 +465,7 @@ constexpr static std::array<SupportedTensorDataTypes, 2> supportedTypeListShape
 constexpr static std::array<SupportedTensorDataTypes, 2> supportedTypeListSize = {SupportedTensorDataTypes::All, SupportedTensorDataTypes::Int64};
 constexpr static std::array<SupportedTensorDataTypes, 1> supportedTypeListQLinearSigmoid = {SupportedTensorDataTypes::UInt8 | SupportedTensorDataTypes::Int8};
 constexpr static std::array<SupportedTensorDataTypes, 2> supportedTypeListAttention = {SupportedTensorDataTypes::Float16to32, SupportedTensorDataTypes::Int32};
+constexpr static std::array<SupportedTensorDataTypes, 2> supportedTypeListGroupNorm = {SupportedTensorDataTypes::Float16to32, SupportedTensorDataTypes::Float16to32};
```
Comment (sumitsays) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp:469]: As per contrib ops doc, gamma and beta (M) support only float. So should we only support float for Tensors of type M?
Please ignore, if I missed anything.

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp [hunk 3]
```diff
@@ -874,6 +877,7 @@ constexpr static OperatorRegistrationInformation operatorRegistrationInformation
     {REG_INFO(      7,  LayerNormalization,                 typeNameListLayerNormContrib,   supportedTypeListLayerNormalizationContrib, DmlGraphSupport::Supported, requiredConstantCpuInputs(), std::nullopt, QueryLayerNormalization)},
     {REG_INFO_MS(   1,  SkipLayerNormalization,             typeNameListDefault,            supportedTypeListFloat16to32,           DmlGraphSupport::Supported)},
     {REG_INFO_MS(   1,  EmbedLayerNormalization,            typeNameListDefault,            supportedTypeListFloat16to32,           DmlGraphSupport::Supported)},
+    {REG_INFO_MS(   1,  GroupNorm,                          typeNameListGroupNorm,          supportedTypeListGroupNorm,             DmlGraphSupport::Supported)},
```
Comment (sumitsays) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp:880]: [nit] Should we put all contrib ops registration under one segment? Either registrations from here to line 852 or vice-versa.

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp (hunk 2 above)
Comment (PatriceVignola) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp:469]: This allows us more leeway in the future if we want to optimize those casts away.

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp (hunk 3 above)
Comment (PatriceVignola) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/OperatorRegistration.cpp:880]: Sure, I'll do it once all my open PRs are done when I implement the next contrib op.

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorGroupNorm.cpp (hunk 1 above)
Comment (PatriceVignola) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorGroupNorm.cpp:27]: Just being overly safe :D basically acting as an assert here.

**Pull request merged**